modules/layers/src/text-layer/multi-icon-layer/multi-icon-layer-fragment.glsl.js

```diff
@@ -36,23 +37,30 @@ varying vec2 uv;
 void main(void) {
   geometry.uv = uv;
 
-  vec4 texColor = texture2D(iconsTexture, vTextureCoords);
-  
-  float alpha = texColor.a;
+  float alpha = texture2D(iconsTexture, vTextureCoords).a;
+
   // if enable sdf (signed distance fields)
   if (sdf) {
-    float distance = texture2D(iconsTexture, vTextureCoords).a;
-    alpha = smoothstep(buffer - vGamma, buffer + vGamma, distance);
+    alpha = smoothstep(buffer - vGamma, buffer + vGamma, alpha);
   }
 
   // Take the global opacity and the alpha from vColor into account for the alpha component
   float a = alpha * vColor.a;
-
+  
   if (a < alphaCutoff) {
-    discard;
+    if (backgroundColor.a < 0.001) {
+      discard;
+    } else {
+      gl_FragColor = vec4(backgroundColor.rgb, backgroundColor.a * vColor.a);
+      return;
```

Comment:
@xintongxia the return statement fixes the issue of highlight.

Reply:
Depending on the value of your backgroundColor, it may mess up picking.

Reply:
You can check `picking_uActive` as a hack.

Reply:
Any way to know if we are in picking mode?

Reply:
OK, I'll do that.